File: Examples2018/UIListViewCustomAdapter/app/src/main/java/course/examples/ui/listlayoutcustomadapter/ListViewActivity.java
package course.examples.ui.listlayoutcustomadapter;

import android.app.ListActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.AdapterView.OnItemClickListener;
import android.widget.TextView;
import android.widget.Toast;

public class ListViewActivity extends ListActivity {
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Create a new Adapter containing a list of colors
        // Set the adapter on this ListActivity's built-in ListView
        setListAdapter(new ListViewAdapter(
                this,
                R.layout.list_item,
                getResources().getStringArray(R.array.colors)));

        // Enable filtering when the user types in the virtual keyboard
        getListView().setTextFilterEnabled(true);

        // Set an setOnItemClickListener on the ListView
        getListView().setOnItemClickListener(new OnItemClickListener() {
            public void onItemClick(AdapterView<?> parent, View view,
                                    int position, long id) {

                TextView textView = view.findViewById(R.id.text);
                // Display a Toast message indicting the selected item
                Toast.makeText(ListViewActivity.this,
                        textView.getText(), Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: ExamplesKotlin/UIListViewCustomAdapter/app/src/main/java/course/examples/ui/listlayoutcustomadapter/ListViewActivity.kt
package course.examples.ui.listlayoutcustomadapter

import android.app.ListActivity
import android.os.Bundle
import android.widget.AdapterView.OnItemClickListener
import android.widget.TextView
import android.widget.Toast

class ListViewActivity : ListActivity() {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Create a new Adapter containing a list of colors
        // Set the adapter on this ListActivity's built-in ListView
        listAdapter = ListViewAdapter(
            this,
            R.layout.list_item,
            resources.getStringArray(R.array.colors)
        )

        listView.setBackgroundColor(resources.getColor(R.color.divider, null))
        // Enable filtering when the user types in the virtual keyboard
        listView.isTextFilterEnabled = true

        // Set an setOnItemClickListener on the ListView
        listView.onItemClickListener = OnItemClickListener { parent, view, position, id ->
            val textView = view.findViewById<TextView>(R.id.text)
            // Display a Toast message indicting the selected item
            Toast.makeText(
                this@ListViewActivity,
                textView.text, Toast.LENGTH_SHORT
            ).show()
        }
    }
}